Signal Build Error with MSVC 2015 / 2017

Building the TileDB Python extension calling out to the superbuild is causing the following error on both  VC2015 and VC2017 toolchains:

build error:

```
      signal_handlers.cc
C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-9wfbd6\build\TileDB-dev\tiledb\sm\global_state\signal_handlers.cc(121): error C2664: 'BOOL SetConsoleCtrlHandler(PHANDLER_ROUTINE,BOOL)': cannot convert argument 1 from 'BOOL (__cdecl *)(DWORD)' to 'PHANDLER_ROUTINE' [C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build9wfbd6\build\TileDB-dev\build\tiledb\tiledb\TILEDB_CORE_OBJECTS.vcxproj
[C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-9wfbd6\build\TileDB-dev\build\tiledb.vcxproj]
C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-9wfbd6\build\TileDB-dev\tiledb\sm\global_state\signal_handlers.cc(121): note: None of the functions with this name in scope match the target type
```

Link:
https://ci.appveyor.com/project/TileDB-Inc/tiledb-py/builds/19859107/job/tm5t6ar28ucvicmi#L473

--- a/tiledb/sm/global_state/signal_handlers.cc
+++ b/tiledb/sm/global_state/signal_handlers.cc
@@ -37,8 +37,8 @@
 #include <signal.h>
 
 #ifdef _WIN32
-#include <Windows.h>
 #include <io.h>
+#include <windows.h>
 #else
 #include <unistd.h>
 #endif
@@ -101,7 +101,7 @@ extern "C" void tiledb_signal_handler(int signum) {
 /*       Win32 implementations       */
 /* ********************************* */
 
-static BOOL win_ctrl_handler(DWORD dwCtrlType) {
+static BOOL WINAPI win_ctrl_handler(DWORD dwCtrlType) {
   switch (dwCtrlType) {
     case CTRL_BREAK_EVENT:
       tiledb_signal_handler(SIGINT);
@@ -118,7 +118,7 @@ Status SignalHandlers::initialize() {
   }
 
   // Win32 applications should also handle Ctrl-Break.
-  if (SetConsoleCtrlHandler(win_ctrl_handler, true) == 0) {
+  if (SetConsoleCtrlHandler(win_ctrl_handler, TRUE) == 0) {
     return Status::Error(std::string("Failed to install Win32 ctrl handler"));
   }
   return Status::Ok();

